tracpro/orgs_ext/middleware.py

```diff
@@ -0,0 +1,32 @@
+from django.utils.translation import ugettext_lazy as _
+from django.http import HttpResponseBadRequest
+
+from temba_client.base import TembaAPIError, TembaConnectionError
+
+
+class HandleTembaAPIError(object):
+    """ Catch all Temba exception errors """
+
+    def process_exception(self, request, exception):
+
+        rapidProConnectionErrorString = _(
+            "RapidPro appears to be down right now or we cannot connect due to your internet connection. "
```

Comment:
I don't think we need to include a remark about the client's internet connection since the server is the one reaching out to RapidPro.